src/data.rs

```diff
@@ -295,6 +295,42 @@ macro_rules! processed_data {
     };
 }
 
+#[derive(Clone, Debug, strum_macros::EnumString, strum_macros::EnumVariantNames)]
+#[strum(serialize_all = "snake_case")]
+pub enum MandatoryData {
+    SystemInfo,
+    CpuUtilization,
+    KernelConfig,
+    Sysctl,
```

Comment:
I'm not sure there is really mandatory data, if we're going to allow users to skip collecting, we might as well let them collect nothing.